pwiz_tools/Skyline/TestPerf/DetectionsPlotTest.cs

```diff
@@ -0,0 +1,215 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Windows.Forms;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using pwiz.Common.Collections;
+using pwiz.SkylineTestUtil;
+using pwiz.Skyline.Controls.Graphs;
+using pwiz.Skyline.Model;
+using ZedGraph;
+
+namespace TestPerf
+{
+    [TestClass]
+    public class DetectionsPlotTest : AbstractFunctionalTestEx
+    {
+        private static readonly int[][] REF_DATA =
+        {
+            new[] {132, 131, 131, 130, 130, 132},       //q = 0.003, Peptides
+            new[] { 142, 141, 141, 140, 140, 141},      //q = 0.003, Precursors
+            new[] { 129, 127, 128, 127, 127, 128 },      //q = 0.001, Peptides
+            new[] { 139, 137, 138, 137, 136, 136 },      //q = 0.001, Precursors
+            new[] { 128, 126, 127, 126, 126, 127 },      //q = 0.001, Peptides, after update
+            new[] { 138, 136, 137, 136, 135, 135 }      //q = 0.001, Precursors, after update
+        };
+
+        private static readonly string[] TIP_TEXT =
+        {
+            "Replicate	2_SW-B",
+            "Precursor Count	137",
+            "Cumulative Count	143",
+            "All Count	133",
+            "-log10 of Q-Value Median	6.0"
+        };
+        private static readonly string[] HISTOGRAM_TIP_TEXT =
+        {
+            "Replicate Count	5",

```

Comment:
These strings are going to be different in different languages, so your test should make use of things like:
Skyline.Properties.Resources.DetectionHistogramPane_Tooltip_ReplicateCount

(Also, the code would probably be easier to understand if you declared the string right were you are using it in the test code)

Reply:
Right. I replaced the strings with resource reference. Not sure about the latter, I think it's better to keep bulky static definitions outside of the main code, unless it carries a performance penalty